sqlmesh/core/engine_adapter/base.py

```diff
@@ -495,6 +495,17 @@ def _create_table_from_columns(
             kwargs: Optional create table properties.
         """
         table = exp.to_table(table_name)
+
+        if not columns_to_types_all_known(columns_to_types):
+            # It is ok if the columns types are not known if the table already exists and IF NOT EXISTS is set
+            if exists and self.table_exists(table_name):
+                return
+            raise SQLMeshError(
+                "Cannot create a table from a self-referencing query without knowing the column types. "
```

Comment:
left a comment in the previous version, but i'm a bit confused by this error message. why does this explicitly call out self-referencing. the method named seems very generic _create_table_from_columns. how do you know this branch is specific to self-referencing queries?